### PR TITLE
Bugfix hook backend detection

### PR DESCRIPTION
**Pull request by BenjamenMeyer**
- Bug Fix: Configuration File did not really have a 'disk' configuration. Updated it to do so.
- Bug Fix: The Hooks seemed to be loaded incorrectly; adjusted usage of loading the hooks through falcon.API() to be more inline with the Falcon documentation; Falcon now takes care of calling the hooks for us
- Enhancement: Updated the driver configurations so that the initial driver spec simply specifies a sub-section of the configuration for which to load the driver
  - f.e deuce.conf.metadata_driver.driver now is either 'sqlite', 'cassandra', or 'mongodb', the actual driver is specified in that section's driver, f.e deuce.conf.metadata_driver.sqlite.driver
  - raises an exception specific to the driver type if the specified driver for metadata or storage is not recognized
- Enhancement: Uses the deuce.conf.block_storage_driver.driver to determine whether to load the Swift Hooks
  - extensible so we could easily add other values as well
- Bug Fix: Updated the configspec.ini to map more of the values from the specific sections
  - these will typically be provided by default if the user copies the example config.ini and most won't need to be changed
- Enhancement: Updated configspec to use specific values for the drivers
- Enhancement: Updated configspec so the cassandra cluster address space is validated as a list of strings

Note: This was PR #252

### CODE REVIEW

**Comment by racklanta** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://ci.cldbckp.com/job/deuce-pullrequest/268/

**Comment by TheSriram** [comment]
:+1: Awesome work, Thanks!

**Comment by jc7998** [comment]
:+1:

File: ini/configspec.ini [hunk 1]
```diff
@@ -9,17 +9,36 @@ datacenter = string
 default_returned_num = integer
 max_returned_num = integer
 [metadata_driver]
+driver = option('sqlite', 'mongodb', 'cassandra')
+    [[sqlite]]
+    driver = string
+	path = string
```
Comment (raxyu) [inline ini/configspec.ini]: Just for my curiosity, is there any special meaning for the different indents in this section?

Comment (BenjamenMeyer) [inline ini/configspec.ini]: don't think so, but would like to know myself too

@TheSriram

Comment (TheSriram) [inline ini/configspec.ini]: Its for defining nested sections AFAIK. http://www.voidspace.org.uk/python/configobj.html#the-config-file-format

Comment (BenjamenMeyer) [inline ini/configspec.ini]: @TheSriram @raxyu To answer Xuan's question:
``Indentation is not significant, but can be preserved. See the description of the indent_type option, in the ConfigObj specifications chapter, for the details.``

In other words, it doesn't matter. The subsections are actually denoted by the square brackets: 

``Any subsections (sections that are inside the current section) are designated by repeating the square brackets before and after the section name. The number of square brackets represents the nesting level of the sub-section. Square brackets may be separated by whitespace; such whitespace, however, will not be present in the output config written by the write method.``

That said, it does help to have them delineated like that to help visually see the difference between the sections. Though most Config/INI formats keep it flat in their examples.

$0.02

Comment (TheSriram) [inline ini/configspec.ini]: ok, in that case, I'm good with either. but having an indentation helps for visual assimilation of data.

**Comment by BenjamenMeyer** [comment]
@dstritzel This merge will affect the config files; should make it easier to manage. Just want you to be aware.

**Pull request merged**